backend/lambda/function.mjs

```diff
@@ -0,0 +1,149 @@
+    };
+
+    try {
+        switch (event.requestContext.resourceId) {
+            case "GET /levels":
+                body = await dynamo.send(
+                    new QueryCommand({
+                        TableName: tableName,
+                        IndexName: "StatusLastUpdatedIndex",
+                        Limit: 10,
+                        KeyConditionExpression: "#status = :status",
+                        ExpressionAttributeNames: {
+                            "#status": "status"
+                        },
+                        ExpressionAttributeValues: {
+                            ":status": "published"
+                        },
+                        ScanIndexForward: false,
+                    })
+                );
+                break;
+            case "GET /levels/{id}":
+                body = await dynamo.send(
+                    new GetCommand({
+                        TableName: tableName,
+                        Key: {
+                            id: event.pathParameters.id,
+                        },
+                    })
+                );
+                if (!body.Item) throw new Error(`Item ${event.pathParameters.id} not found`);
+                body = body.Item;
+                break;
+            case "POST /levels":
+                const requestData = JSON.parse(event.body);
+                let generatedLevelId = uuidv4();
+                await dynamo.send(
+                    new PutCommand({
+                        TableName: tableName,
+                        Item: {
+                            id: generatedLevelId,
+                            lastUpdated: Date.now().toString(),
+                            name: requestData.name,
+                            creator: requestData.username,
+                            status: requestData.status,
+                            levelData: requestData.levelData,
+                        },
+                    })
+                );
+                body = `Created new level ${generatedLevelId}`;
+                break;
+            case "PUT /levels/{id}":
+                let requestJSON = JSON.parse(event.body);
+                await dynamo.send(
+                    new PutCommand({
+                        TableName: tableName,
+                        Item: {
+                            id: event.pathParameters.id,
+                            lastUpdated: Date.now().toString(),
+                            name: requestJSON.name,
+                            status: requestJSON.status,
+                            creator: requestData.username,
+                            levelData: requestJSON.levelData,
+                        },
+                    })
+                );
+                body = `Updated level ${event.pathParameters.id}`;
+                break;
+            case "DELETE /levels/{id}":
+                // We dont actually want to delete so lets just unpublish.
```

Comment:
Ah right, yeah I was going to take some time to more thoroughly define what these APIs should be/look like https://github.com/LPGameDevs/EditarrrPublic/issues/53 and probably change them a bit (or maybe a lot)

Reply:
So I was about to add the dynamodb permission for DELETE and I thought... hmmm maybe we dont want that option? I then went on a brief tangent doing a lot of querying and decided against fixing everything now. We can update/define the requirements later :)